terraform/modules/neptune-cluster-parameter-group/locals.tf

```diff
@@ -0,0 +1,3 @@
+locals {
```

Comment:
can resource_prefix be a passed variable rather than a generated string - this can have a default value, but we need to make sure that we aren't creating a naming dependency in the modules. these should be as configurable as possible while defining the desired default values